packages/dashboard/src/components/DashboardFrame/index.tsx

```diff
@@ -46,20 +117,57 @@ export interface PageFrameProps extends React.PropsWithChildren<{}> {
     title: React.ReactNode | string
     primaryAction?: React.ReactNode
 }
+
 export function PageFrame(props: PageFrameProps) {
     const classes = useStyles()
     const left = typeof props.title === 'string' ? <Typography variant="h6">{props.title}</Typography> : props.title
     const right = props.primaryAction
+    const matches = useMediaQuery<Theme>((theme) => theme.breakpoints.down(1184))
+
+    const [open, setOpen] = useState(false)
+
+    useEffect(() => {
+        setOpen(false)
+    }, [matches])
+
     return (
         <>
             <AppBar position="relative" color="inherit" elevation={0}>
-                <Toolbar component={Grid} container>
-                    {left}
-                    <Box sx={{ flex: 1 }} />
-                    {right}
+                <Toolbar component={Grid} container className={classes.toolbar}>
+                    {matches && (
+                        <Grid item xs={2} container alignItems="center" className={classes.logo}>
+                            <IconButton onClick={() => setOpen(!open)} className={classes.menuButton}>
+                                {open ? <CloseIcon /> : <MenuIcon />}
+                            </IconButton>
+
+                            <img height={40} alt="Mask Logo" src="https://mask.io/assets/icons/logo.svg" />
+                        </Grid>
+                    )}
+                    <Grid item xs={matches ? 10 : 12} container className={classes.title}>
+                        {left}
+                        <Box sx={{ flex: 1 }} />
+                        {right}
+                    </Grid>
                 </Toolbar>
             </AppBar>
             <Grid item xs className={classes.containment}>
+                {matches && (
+                    <Drawer
+                        open={open}
+                        onClose={() => setOpen(false)}
+                        BackdropProps={{ invisible: true }}
+                        PaperProps={{ elevation: 0 }}
+                        variant="temporary"
+                        ModalProps={{
+                            BackdropProps: {
+                                classes: { root: classes.drawerBackdropRoot },
+                            },
+                        }}
+                        // className={classes.temporaryDrawer}
+                        classes={{ paper: classes.temporaryPaper, root: classes.drawerRoot }}>
+                        <Navigation />
```

Comment:
So I found that on the small screen the state of Navigation still lost. Maybe we should use context to preserve that?